testrunner-running/src/main/scala/edu/illinois/cs/statecapture/StateCapture.java

```diff
@@ -0,0 +1,755 @@
+            error.printStackTrace();
+            dirty = true;
+        }
+        return s;
+    }
+
+    private String serializeOBs(Object ob) {
+        XStream xstream = getXStreamInstance();
+        String s = "";
+
+        try {
+            s = xstream.toXML(ob);
+            s = sanitizeXmlChars(s);
+        } catch (Exception e) {
+            // In case serialization fails, mark the StateCapture for this test
+            // as dirty, meaning it should be ignored
+            dirty = true;
+            throw e;
+        }
+        return s;
+    }
+
+    private static class AlphabeticalFieldkeySorter implements FieldKeySorter {
+        @Override
+        public Map sort(Class type, Map keyedByFieldKey) {
+            final Map<FieldKey, Field> map = new TreeMap<>(new Comparator<FieldKey>() {
+
+                @Override
+                public int compare(final FieldKey fieldKey1, final FieldKey fieldKey2) {
+                    return fieldKey1.getFieldName().compareTo(fieldKey2.getFieldName());
+                }
+            });
+            map.putAll(keyedByFieldKey);
+            return map;
+        }
+    }
+
+    private XStream getXStreamInstance() {
+        XStream xstream = new XStream(JVM.newReflectionProvider(new FieldDictionary(
+                new AlphabeticalFieldkeySorter())),new DomDriver());
+
+
+        xstream.setMode(XStream.XPATH_ABSOLUTE_REFERENCES);
+        xstream.addPermission(AnyTypePermission.ANY);
+        // Set fields to be omitted during serialization
+
+        xstream.omitField(Thread.class, "contextClassLoader");
+        xstream.omitField(java.security.ProtectionDomain.class, "classloader");
+        xstream.omitField(java.security.ProtectionDomain.class, "codesource");
+        xstream.omitField(ClassLoader.class, "defaultDomain");
+        xstream.omitField(ClassLoader.class, "classes");
+
+        xstream.omitField(java.lang.ref.SoftReference.class, "timestamp");
+        xstream.omitField(java.lang.ref.SoftReference.class, "referent");
+        xstream.omitField(java.lang.ref.Reference.class, "referent");
+
+        xstream.registerConverter(new CustomMapConverter(xstream.getMapper()));
+
+        for (String ignore : ignores) {
+            int lastDot = ignore.lastIndexOf(".");
+            String clz = ignore.substring(0,lastDot);
+            String fld = ignore.substring(lastDot+1);
+            try {
+                xstream.omitField(Class.forName(clz), fld);
+            } catch (Exception ex) {
+                ex.printStackTrace();
+            }
+        }
+
+        return xstream;
+    }
+    
+    protected String getFieldFQN(Field f) {
+        String clz = f.getDeclaringClass().getName();
+        String fld = f.getName();
+        return clz + "." + fld;
+    }
+
+    public String readFile(String path) throws IOException {
+        File file = new File(path);
+        return FileUtils.readFileToString(file, "UTF-8");
+    }
+
+    int countDirNums(String path) {
```

Comment:
Probably don't need this helper method anymore if we don't use to generate file names.